services/thermal_mgr/thermal_mgr.c

```diff
@@ -17,55 +18,107 @@ static StackType_t thermalMgrTaskStack[THERMAL_MGR_STACK_SIZE];
 
 #define THERMAL_MGR_QUEUE_LENGTH 10U
 #define THERMAL_MGR_QUEUE_ITEM_SIZE sizeof(thermal_mgr_event_t)
+#define THERMAL_MGR_HYSTERESIS_TEMP 75U
+#define THERMAL_MGR_OVERTEMP 80U
 
 static QueueHandle_t thermalMgrQueueHandle;
 static StaticQueue_t thermalMgrQueueBuffer;
 static uint8_t thermalMgrQueueStorageArea[THERMAL_MGR_QUEUE_LENGTH * THERMAL_MGR_QUEUE_ITEM_SIZE];
 
 static void thermalMgr(void *pvParameters);
 
-void initThermalSystemManager(lm75bd_config_t *config) {
+void initThermalSystemManager(lm75bd_config_t *config)
+{
   memset(&thermalMgrTaskBuffer, 0, sizeof(thermalMgrTaskBuffer));
   memset(thermalMgrTaskStack, 0, sizeof(thermalMgrTaskStack));
-  
+
   thermalMgrTaskHandle = xTaskCreateStatic(
-    thermalMgr, "thermalMgr", THERMAL_MGR_STACK_SIZE,
-    config, 1, thermalMgrTaskStack, &thermalMgrTaskBuffer);
+      thermalMgr, "thermalMgr", THERMAL_MGR_STACK_SIZE,
+      config, 1, thermalMgrTaskStack, &thermalMgrTaskBuffer);
 
   memset(&thermalMgrQueueBuffer, 0, sizeof(thermalMgrQueueBuffer));
   memset(thermalMgrQueueStorageArea, 0, sizeof(thermalMgrQueueStorageArea));
 
   thermalMgrQueueHandle = xQueueCreateStatic(
-    THERMAL_MGR_QUEUE_LENGTH, THERMAL_MGR_QUEUE_ITEM_SIZE,
-    thermalMgrQueueStorageArea, &thermalMgrQueueBuffer);
-
+      THERMAL_MGR_QUEUE_LENGTH, THERMAL_MGR_QUEUE_ITEM_SIZE,
+      thermalMgrQueueStorageArea, &thermalMgrQueueBuffer);
 }
 
-error_code_t thermalMgrSendEvent(thermal_mgr_event_t *event) {
+error_code_t thermalMgrSendEvent(thermal_mgr_event_t *event)
+{
   /* Send an event to the thermal manager queue */
-
+  error_code_t errCode;
+  if (!event)
+  {
+    return ERR_CODE_INVALID_ARG;
+  }
+  if (!thermalMgrQueueHandle)
+  {
+    return ERR_CODE_QUEUE_HANDLE_NULL;
+  }
+  if (uxQueueSpacesAvailable(thermalMgrQueueHandle) == 0)
```

Comment:
This isn't good practice as this could succeed but then this task can get pre-empted by another task that sends another event to the queue. Thus the send below will fail for the original function